internal/server/evaluator.go

```diff
@@ -130,7 +130,7 @@ func (s *Server) evaluate(ctx context.Context, r *flipt.EvaluationRequest) (resp
 	for _, rule := range rules {
 		if rule.Rank < lastRank {
 			resp.Reason = flipt.EvaluationReason_ERROR_EVALUATION_REASON
-			return resp, errs.ErrInvalidf("rule rank: %d detected out of order", rule.Rank)
```

Comment:
not a fan of these helper functions?

Reply:
Ahh yeah this was a suggestion I snuck in.
I noticed that every time we added a new `type SomeError string` we ended up having to make a helper `f`.
This was a proposal to just have a single `NewErrorf` which could still be a one line but meant we don't have to add a new extra method every time.

What do you think? I'm not married to it if this is worse. It is a little longer at each call-site.
I guess you could argue this way isn't much better than `errors.SomeError(fmt.Sprintf("...", ...))`.

Reply:
Nice. Love it.